hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/DataChecksum.java

```diff
@@ -78,6 +100,16 @@ public static Checksum newCrc32() {
     return new CRC32();
   }
 
+  public static Checksum newCrc32C() {
+    try {
+      return Shell.isJava9OrAbove() ? (Checksum) NEW_CRC32C_MH.invoke()
+              : new PureJavaCrc32C();
+    } catch (Throwable e) {
+      // Should not reach here.
+      throw new RuntimeException(e);
```

Comment:
RuntimeException shouldn't be wrapped; leave the other Throwables alone. So: `catch(RuntimeException e) { throw e;}`, then catch and wrap Exception only,

Reply:
Got it, this will be more clean.